### ISSUE TITLE
Clang frontend command failed to build drivers/vhost/vringh.c in Linux kernel

### ISSUE THREAD

**Issue by eve2023** [llvm/llvm-project#82854]
```
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: clang -Wp,-MMD,drivers/vhost/.vringh.o.d -nostdinc -I./arch/x86/include -I./arch/x86/include/generated -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/compiler-version.h -include ./include/linux/kconfig.h -include ./include/linux/compiler_types.h -D__KERNEL__ --target=x86_64-linux-gnu -fintegrated-as -Werror=unknown-warning-option -Werror=ignored-optimization-argument -Werror=option-ignored -Werror=unused-command-line-argument -fmacro-prefix-map=./= -std=gnu11 -fshort-wchar -funsigned-char -fno-common -fno-PIE -fno-strict-aliasing -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -fcf-protection=none -m64 -falign-loops=1 -mno-80387 -mno-fp-ret-in-387 -mstack-alignment=8 -mskip-rax-setup -mtune=generic -mno-red-zone -mcmodel=kernel -Wno-sign-compare -fno-asynchronous-unwind-tables -mretpoline-external-thunk -mindirect-branch-cs-prefix -mfunction-return=thunk-extern -mharden-sls=all -fpatchable-function-entry=16,16 -fno-delete-null-pointer-checks -O2 -fstack-protector-strong -fno-omit-frame-pointer -fno-optimize-sibling-calls -ftrivial-auto-var-init=zero -fno-stack-clash-protection -fzero-call-used-regs=used-gpr -pg -mfentry -DCC_USING_NOP_MCOUNT -DCC_USING_FENTRY -falign-functions=16 -fstrict-flex-arrays=3 -fno-strict-overflow -fno-stack-check -Wall -Wundef -Werror=implicit-function-declaration -Werror=implicit-int -Werror=return-type -Werror=strict-prototypes -Wno-format-security -Wno-trigraphs -Wno-frame-address -Wno-address-of-packed-member -Wmissing-declarations -Wmissing-prototypes -Wframe-larger-than=1024 -Wno-gnu -Wno-unused-but-set-variable -Wno-unused-const-variable -Wvla -Wno-pointer-sign -Wcast-function-type -Wimplicit-fallthrough -Werror=date-time -Werror=incompatible-pointer-types -Wenum-conversion -Wno-unused-but-set-variable -Wno-unused-const-variable -Wno-format-overflow -Wno-format-truncation -Wno-pointer-to-enum-cast -Wno-tautological-constant-out-of-range-compare -Wno-unaligned-access -Wno-cast-function-type-strict -Wno-missing-field-initializers -Wno-type-limits -Wno-shift-negative-value -Wno-initializer-overrides -Wno-sign-compare -g -gdwarf-5 -fsanitize=array-bounds -fsanitize=shift -fsanitize=bool -fsanitize=enum -DMODULE -DKBUILD_BASENAME=\"vringh\" -DKBUILD_MODNAME=\"vringh\" -D__KBUILD_MODNAME=kmod_vringh -c -o drivers/vhost/vringh.o drivers/vhost/vringh.c
1.	<eof> parser at end of file
2.	Optimizer
 #0 0x000055b618d6cdc8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2d94dc8)
 #1 0x000055b618d6abae llvm::sys::RunSignalHandlers() (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2d92bae)
 #2 0x000055b618ce8669 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #3 0x00007f01d7042990 (/lib/x86_64-linux-gnu/libc.so.6+0x42990)
 #4 0x000055b618824e39 llvm::DebugInfoFinder::processScope(llvm::DIScope*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x284ce39)
 #5 0x000055b6188251ff llvm::DebugInfoFinder::processVariable(llvm::Module const&, llvm::DILocalVariable const*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x284d1ff)
 #6 0x000055b618825318 llvm::DebugInfoFinder::processDPValue(llvm::Module const&, llvm::DPValue const&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x284d318)
 #7 0x000055b618824bfb llvm::DebugInfoFinder::processInstruction(llvm::Module const&, llvm::Instruction const&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x284cbfb)
 #8 0x000055b618db2336 llvm::CloneBasicBlock(llvm::BasicBlock const*, llvm::ValueMap<llvm::Value const*, llvm::WeakTrackingVH, llvm::ValueMapConfig<llvm::Value const*, llvm::sys::SmartMutex<false>>>&, llvm::Twine const&, llvm::Function*, llvm::ClonedCodeInfo*, llvm::DebugInfoFinder*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2dda336)
 #9 0x000055b618db2b67 llvm::CloneFunctionInto(llvm::Function*, llvm::Function const*, llvm::ValueMap<llvm::Value const*, llvm::WeakTrackingVH, llvm::ValueMapConfig<llvm::Value const*, llvm::sys::SmartMutex<false>>>&, llvm::CloneFunctionChangeType, llvm::SmallVectorImpl<llvm::ReturnInst*>&, char const*, llvm::ClonedCodeInfo*, llvm::ValueMapTypeRemapper*, llvm::ValueMaterializer*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2ddab67)
#10 0x000055b618db3acc llvm::CloneFunction(llvm::Function*, llvm::ValueMap<llvm::Value const*, llvm::WeakTrackingVH, llvm::ValueMapConfig<llvm::Value const*, llvm::sys::SmartMutex<false>>>&, llvm::ClonedCodeInfo*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2ddbacc)
#11 0x000055b619e84108 llvm::FunctionSpecializer::createSpecialization(llvm::Function*, llvm::SpecSig const&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x3eac108)
#12 0x000055b619e83180 llvm::FunctionSpecializer::run() (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x3eab180)
#13 0x000055b619e63c82 llvm::IPSCCPPass::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x3e8bc82)
#14 0x000055b619cb18cd llvm::detail::PassModel<llvm::Module, llvm::IPSCCPPass, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x3cd98cd)
#15 0x000055b6188f1554 llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2919554)
#16 0x000055b618f1a71c (anonymous namespace)::EmitAssemblyHelper::RunOptimizationPipeline(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>&, std::unique_ptr<llvm::ToolOutputFile, std::default_delete<llvm::ToolOutputFile>>&, clang::BackendConsumer*) BackendUtil.cpp:0:0
#17 0x000055b618f132a4 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>, clang::BackendConsumer*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2f3b2a4)
#18 0x000055b61939f537 clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x33c7537)
#19 0x000055b61aa31766 clang::ParseAST(clang::Sema&, bool, bool) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x4a59766)
#20 0x000055b61961e306 clang::FrontendAction::Execute() (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x3646306)
#21 0x000055b619593b64 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x35bbb64)
#22 0x000055b6196daf0e clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x3702f0e)
#23 0x000055b616e13ffc cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0xe3bffc)
#24 0x000055b616e11001 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&, llvm::ToolContext const&) driver.cpp:0:0
#25 0x000055b6193e96a9 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const::$_0>(long) Job.cpp:0:0
#26 0x000055b618ce844c llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x2d1044c)
#27 0x000055b6193e90f6 clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x34110f6)
#28 0x000055b6193ab5b7 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x33d35b7)
#29 0x000055b6193aba5e clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x33d3a5e)
#30 0x000055b6193c9dcf clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0x33f1dcf)
#31 0x000055b616e106c8 clang_main(int, char**, llvm::ToolContext const&) (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0xe386c8)
#32 0x000055b616e1eca7 main (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0xe46ca7)
#33 0x00007f01d7028150 __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:74:3
#34 0x00007f01d7028209 call_init ./csu/../csu/libc-start.c:128:20
#35 0x00007f01d7028209 __libc_start_main ./csu/../csu/libc-start.c:347:5
#36 0x000055b616e0d845 _start (/home/eve/Downloads/llvm-project/build2/bin/clang-19+0xe35845)
clang: error: clang frontend command failed with exit code 139 (use -v to see invocation)
clang version 19.0.0git (https://github.com/llvm/llvm-project.git 775bd60363353b78657967c80f0f109cdb65cf8f)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/eve/Downloads/llvm-project/build2/bin
clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /tmp/vringh-217e0f.c
clang: note: diagnostic msg: /tmp/vringh-217e0f.sh
clang: note: diagnostic msg: 

********************
```
[vringh-217e0f.zip](https://github.com/llvm/llvm-project/files/14391104/vringh-217e0f.zip)
This issue occurs at commit:
```
commit 775bd60363353b78657967c80f0f109cdb65cf8f (HEAD, origin/main, origin/HEAD)
Author: Visoiu Mistrih Francis <890283+francisvm@users.noreply.github.com>
Date:   Fri Feb 23 15:44:57 2024 -0800
```
This issue does not occur at commit:
```
commit e09e0d52a03c7141a7d62fb4adf4d9fee32bebb8 (HEAD)
Author: tsitdikov <tsitdikov@google.com>
Date:   Fri Feb 23 12:54:15 2024 +0000
```

**Comment by eve2023** [comment]
The crash no longer reproduces at 1e98d4

**Comment by nathanchance** [comment]
You may want to consider building with `-DLLVM_ENABLE_ASSERTIONS=ON` if you are going to be testing with tip of tree a lot. While it will make your toolchain slower due to extra checks, your reports will have higher quality due to better messages when a crash happens, which also gives you the ability to search for if the problem has already been reported and fixed.

**Comment by OCHyams** [comment]
Thanks for reporting the issue.

> Would be great to know the cause, though.

Looking at the stack trace I believe it is the same crash mentioned in comments on 71d47a0b00e9f48dc740556d7f452ffadf308731, which was then reverted. Should be fixed with [6067711](https://github.com/llvm/llvm-project/commit/60677110a2e0b7ba9f5bf2fcfc1b3f8888eb7f58) (reapplied original patch in 3356818eed3224c50012f8ed2bfa046f2bc8e154).
